api/src/main/java/com/netflix/iceberg/Schema.java

```diff
@@ -207,13 +211,31 @@ public Schema select(String... names) {
    * @return a projection schema from this schema, by name
    */
   public Schema select(Collection<String> names) {
+    return select(names, true);
+  }
+
+  /**
+   * Creates a projection schema for a subset of columns, selected by name, and with the specified case sensitivity.
+   * <p>
+   * Names that identify nested fields will select part or all of the field's top-level column.
+   *
+   * @param names a List of String names for selected columns
+   * @return a projection schema from this schema, by name
+   */
+  public Schema select(Collection<String> names, boolean caseSensitive) {
```

Comment:
Re this method signature, we could instead do the refinement pattern we've done in other places like so:

`schema.caseSensitive(false).select(..)`

But I'm not sure refinements should be made against `Schema` objects as it feels that they should strictly represent what the underlying schema is. Otherwise it could get confusing if you do a `schema.caseSensitive(false).asStruct()`.. should that return all fields lowercase or not? Thus decided to not do refinement pattern and instead overload the `select()` method.

LMK what you think @rdblue.

Reply:
I think it would be better to have separate methods. It is more clear what's happening that way.

The schema case should match the table case. If the user requests a case-insensitive projection, then it is fine if Iceberg chooses the case of the result. Matching the table makes the most sense because it is a projection selected by these names, not a projection created from the names.

Reply:
What about using a different method name? This implementation could be `internalSelect` and we could add a public `caseInsensitiveSelect`. I guess we could also expose `internalSelect` as package-private and defer how to implement this in the public API until later.

Reply:
Ah, I like the approach of `caseInsensitiveSelect`. A public API with specific semantics like that would be useful, and not make follow up calls confusing like the refinement pattern. Will go with that!

Reply:
Done.